code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm

```diff
@@ -70,6 +70,10 @@ Difficulty: Medium
 	. = ..()
 	internal = new/obj/item/gps/internal/dragon(src)
 
+/mob/living/simple_animal/hostile/megafauna/dragon/death()
+	qdel(internal) // so drake corpses don't have a gps signal
```

Comment:
I think you can just take these off the global list temporarily.

It's an edge case but It would lose the gps if it gets revived.